pkg/client/typed/scale/scale.go

```diff
@@ -0,0 +1,218 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scale
+
+import (
+	"fmt"
+	"reflect"
+
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/client/typed/dynamic"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+type Client struct {
+	dc *dynamic.Client
+}
+
+type ScaleInterface interface {
+	Replicas() (int, error)
+	SetReplicas(replicas int)
+	Selector() (string, error)
+}
+
+func NewClient(conf *restclient.Config) (*Client, error) {
+	// Make a copy, do not modify the passed config.
+	// TODO(madhusudancs): Should we just receive the conf argument by value? Receiving this as a
+	// pointer now just to be consistent with other clients.
+	conf2 := *conf
+	if conf2.APIPath == "" {
+		conf2.APIPath = "/apis"
```

Comment:
> The whole point of this is to allow dynamic determination of where to send the scale request

@deads2k yeah, but how do you determine that here in the constructor? 

> and this fails on the very first thing it has to scale: replicationcontrollers, right?

Well, the very first thing it has to scale is (arguably?) /apis/extensions/v1beta1/.../replicationcontrollers/.../scale :-) And the other candidates are /apis/extensions/v1beta1/.../{deployments,replicasets}.

@liggitt Sure `/api/v1/replicationcontrollers/scale` is an eligible candidate, but should that be the default? 

The reason why I chose /apis as the default is because that's wins by majority, at least right now. And then if you want /api/v1/... or any other prefixes, then you need to pass your own prefix while constructing/instantiating the client. 

I find neither arguments (to make /api or /apis as default) compelling. At this point, quite honestly, I feel that we are just bike shedding. 

> Alternatively, you could take a subresourceref as input to this client's constructor. But that might be inconvenient.

It certainly is inconvenient.

One alternative, if @krousey agrees, is to not set the APIPath prefix to a default value at all and then call the Request.Prefix() method to set it when we have more information in Get, Update, etc. methods.